test/built-ins/TypedArray/prototype/copyWithin/bit-precision.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-%typedarray%.prototype.copywithin
+es6id: 22.2.3.5
+description: Preservation of bit-level encoding
+info: |
+  Array.prototype.copyWithin (target, start [ , end ] )
+
+  12. Repeat, while count > 0
+    [...]
+    d. If fromPresent is true, then
+      i. Let fromVal be ? Get(O, fromKey).
+      ii. Perform ? Set(O, toKey, fromVal, true).
+includes: [nans.js, compareArray.js, testTypedArray.js]
+---*/
+
+var distinctNaNsDouble = new Array(distinctNaNs.length * 2);
+
+distinctNaNs.forEach(function(v, i) {
+  distinctNaNsDouble[i] = v;
+});
+
+function body(FloatArray) {
+  var subject = new FloatArray(distinctNaNsDouble);
```

Comment:
Possible further simplification: Rather than make a distinctNaNsDouble array, what if you allocated a FloatArray of the appropriate size and wrote each distinct NaN into the first half?